Define what a "switch" is

nielsbom opened the pull request:
The term "switch" is used exclusively here, "flag" is more common I think. I added some prose in the beginning to try and make this more clear.

Wikipedia has some (hopefully) default language for this: https://en.wikipedia.org/wiki/Command-line_interface#Command-line_option

josevalim commented on lib/elixir/lib/option_parser.ex:
```diff
@@ -1,9 +1,15 @@
-  developers to parse a list of arguments into options:
+  Functions for parsing command line arguments.
+
+  When calling a command, it's possible to pass commandline options
```
```suggestion
  When calling a command, it's possible to pass command line options
```

josevalim commented on lib/elixir/lib/option_parser.ex:
```diff
@@ -1,9 +1,15 @@
+  When calling a command, it's possible to pass commandline options
+  to modify what that command does. In this documentation those are
+  called "switches", in other situations they may be called "flags"
+  or simply "options". A switch can be given a value, also called an
```
I don't think use "arguments" to refer to switch values in these docs though, do we?

nielsbom replied:
Looks like we do that here: https://hexdocs.pm/elixir/OptionParser.html#parse/2-types
"Switches parsed by OptionParser may take zero or one arguments."

nielsbom commented:
I think there's quite a lot of (mixed) terminology in these docs, which is understandable, but also confusing :-)
We parse "command line arguments" into "options" but we call those "switches"... right?

Thinking about this again, we (I) could add an example in the beginning to define terminology...

josevalim commented:
@nielsbom that's a good point. By trying to define what it isn't, it makes it more confusing. Maybe we should just focus on the definition:

> When calling a command, it's possible to modify its behaviour by
> passing "switches" via the command line. A switch starts with two
> dashes (`-`) and is followed by alphanumeric characters, dashes,
> and underscore, such as `--foobar`. After a switch, zero, one or
> multiple arguments may be given. In other literate, switches may
> be referred to as "flags" or simply "options".

Thoughts?

nielsbom commented:
I like examples so I took your suggestion and added some examples:

> When calling a command, it's possible to modify its behaviour by
passing one or more "switches" via the command line. A switch starts with two
dashes (-) and the name of the switch. Instead of the full name it's also possible to define and use a one-character alias of that switch, preceded by a single dash (-). 
For example, in `elixir --version` the `version` switch is used. This switch has an alias that is used like so: `elixir -v`.
After a switch, zero, one or multiple arguments may be given.
For example, in `iex -S mix` we use the `S` switch and pass it the value of `mix`.
In other systems, switches may be referred to as "flags" or simply "options".

I removed the "alphanumeric characters, dashes, and underscore" because I think that's too much detail at this time in the text.

josevalim commented:
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
